draft-ietf-quic-transport.md

```diff
@@ -594,6 +596,27 @@ supports, as shown below.
 ~~~
 {: #version-negotiation-format title="Version Negotiation Packet"}
 
+A server that generates a version negotiation packet SHOULD include one of the
+reserved version numbers (those following the pattern 0x?a?a?a?a).  This ensures
+that clients are able to correctly handle an unsupported version.
+
+The design of version negotiation permits a server to avoid maintaining state
+for packets that it rejects in this fashion.  However, when the server generates
+a version negotiation packet, it cannot randomly generate a reserved version
+number.  This is because the server is required to include the same value in its
+transport parameters.  To avoid the selected version number changing during
+connection establishment, the reserved version can be generated as a function of
+values that will be available to the server when later generating its handshake
+packets.
+
+A pseudorandom function that takes client address information (IP and port) and
+the client selected version as input would ensure that there is sufficient
+variability in the values that a server uses.
+
+A client MAY send a packet using a reserved version number.  This can be used to
+solicit a list of supported versions from a server.
```

Comment:
If you are creating a connection in response to Alt-Svc, you aren't depending on the connection establishing in any particular time.  Thus, it might make sense to do this with some low probability just to keep the servers honest.